internal/scanner/walk.go

```diff
@@ -166,6 +169,16 @@ func (w *Walker) walkAndHashFiles(fchan chan protocol.FileInfo) filepath.WalkFun
 			return skip
 		}
 
+		if w.Pruner != nil {
```

Comment:
I don't think it's an anti-pattern... In this case, I think it makes sense for `f.ShouldSkip()` to be `false` when `f` is `nil` (rather than an error), much like `len(f)` is zero for nil slices.